tst/functions/Mock.Tests.ps1

```diff
@@ -2798,6 +2798,79 @@ Describe 'Mocking using ParameterFilter' {
     }
 }
 
+Describe 'Calls not matching ParameterFilter' {
+    BeforeAll {
+        function demo ($name) {
+            "hello $name"
+        }
+    }
+
+    Context 'When default mock does not exist in parent scope' {
+        It 'Throws by default' {
+            Mock demo -ParameterFilter { $name -eq 'world' } -MockWith { 'mocked' }
+            { demo -name 'you' } | Should -Throw
+        }
+
+        It 'Calls locally defined default mock' {
+            Mock demo -ParameterFilter { $name -eq 'world' } -MockWith { 'mocked' }
+            Mock demo { 'default mock' }
+            demo -name 'you' | Should -Be 'default mock'
+        }
+
+        It 'Calls real function when -AllowFallback is used' {
+            Mock demo -ParameterFilter { $name -eq 'world' } -MockWith { 'mocked' } -AllowFallback
+            demo -name 'you' | Should -Be 'hello you'
+        }
+
+        It 'Calls real function when at least one parameterized mock has -AllowFallback' {
+            # TODO: Is this expected? Or should all parameterized mocks allow fallback?
+            Mock demo -ParameterFilter { $name -eq 'world' } -MockWith { 'mocked' } -AllowFallback
+            Mock demo -ParameterFilter { $name -eq 'Wisconsin' } -MockWith { 'mocked2' }
+            demo -name 'you' | Should -Be 'hello you'
+        }
```

Comment:
How should this work? This is where `-AllowFallback/AllowOriginal/AllowPassthrough` gets ugly compared to having a `Mock demo -Throw` shortcut to implement a guard default mock.